### PR TITLE
Added support for SNAT in Virtual Server CRD

### PR DESCRIPTION
**Pull request by vklohiya**

### CODE REVIEW

File: pkg/crmanager/resourceConfig.go [hunk 1]
```diff
@@ -387,6 +389,12 @@ func (crMgr *CRManager) prepareRSConfigFromVirtualServer(
 	rsCfg.Pools = append(rsCfg.Pools, pools...)
 	rsCfg.Monitors = append(rsCfg.Monitors, monitors...)
 
+	// set the SNAT policy to auto is it's not defined by end user
+	if !(len(vs.Spec.SNAT) > 0) {
```
Comment (agupta49) [inline pkg/crmanager/resourceConfig.go:393]: This should be vs.Spec.SNAT==""

Comment (vklohiya) [reply pkg/crmanager/resourceConfig.go]: Update the code.

File: pkg/crmanager/backend.go [hunk 2]
```diff
@@ -386,6 +386,13 @@ func createServiceDecl(cfg *ResourceConfig, sharedApp as3Application) {
 	svc.TranslateServerAddress = true
 	svc.TranslateServerPort = true
 
+	if cfg.Virtual.SNAT == "auto" || cfg.Virtual.SNAT =="none" {
```
Comment (agupta49) [inline pkg/crmanager/backend.go:389]: This code block is not formatted properly.

Comment (vklohiya) [reply pkg/crmanager/backend.go]: Addressed formatting issue

**Review by agupta49 — APPROVED** [review]

**Pull request merged**